File: 110-balanced-binary-tree/balanced-binary-tree.py
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def height(root):
            if not root:
                return 0
            return 1 + max(height(root.left), height(root.right))
        if not root:
            return True
        return (
            (abs(height(root.left) - height(root.right)) < 2)
            and self.isBalanced(root.left)
            and self.isBalanced(root.right)
        )

File: 110-balanced-binary-tree/balanced-binary-tree.java

class Solution {
    public boolean isBalanced(TreeNode root) {
        return isbal(root) != -1;
    }

    int isbal(TreeNode root) {
        if (root == null) return 0;

        int lh = isbal(root.left);
        if (lh == -1) return -1;

        int rh = isbal(root.right);
        if (rh == -1) return -1;

        int temp=Math.abs(lh - rh);
        if (temp > 1) 
            return -1;
        else
            return Math.max(lh, rh) + 1;
    }
}